app/services/catalog/copy_portfolio_item.rb

```diff
@@ -0,0 +1,70 @@
+module Catalog
+  class CopyPortfolioItem
+    attr_reader :new_portfolio_item
+
+    def initialize(params)
+      @name = params[:portfolio_item_name]
+      @portfolio_item = PortfolioItem.find(params[:portfolio_item_id])
+
+      begin
+        @to_portfolio = Portfolio.find(params[:portfolio_id] || @portfolio_item.portfolio_id)
+      rescue ActiveRecord::RecordNotFound
+        raise Catalog::InvalidParameter, "Portfolio specified not found"
+      end
+    end
+
+    def process
+      @new_portfolio_item = make_copy
+      @to_portfolio.portfolio_items << @new_portfolio_item
```

Comment:
Do we even need to make this call? The `@new_portfolio_item` is getting the `@to_portfolio.id` set, so the relationship is being saved there, I think?

Reply:
Yes we actually do need to, if the user passes in a `portfolio_id` then the portfolio will be different than the one we copied from.